.github/workflows/ci.yml

```diff
@@ -368,3 +368,57 @@ jobs:
           else
             echo "Integration tests passed"
           fi
+  smoke-test-released-exemplar:
+    name: Smoke Test Against Released Exemplar
+    if: github.event_name != 'pull_request' || github.event.action == 'enqueued' || contains( github.event.pull_request.labels.*.name, 'run-all')
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the repository with full history
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.head_ref }}
+          fetch-depth: 0
+          fetch-tags: true
+      - name: Get latest release tag
+        run: |
+          latest_release=$(git tag --sort=-v:refname | grep -v v1 | head -n 1)
+          echo "Latest release: $latest_release"
+          echo "LATEST_RELEASE=$latest_release" >> "$GITHUB_ENV"
+      - name: Check out exemplar from previous release
+        uses: actions/checkout@v4
+        with:
+          clean: false
+          repository: TBD54566975/ftl.git
+          path: tmpsmoketest
+          ref: ${{ env.LATEST_RELEASE }}
+          fetch-depth: 1
+          fetch-tags: true
+          sparse-checkout: smoketest
+          sparse-checkout-cone-mode: false
+      - name: Init Hermit
+        uses: cashapp/activate-hermit@v1
+        with:
+          cache: true
+      - name: Build Cache
+        uses: ./.github/actions/build-cache
+      - name: Docker Compose
+        run: docker compose up -d --wait
+      - name: Create DB
+        run: just init-db
+      - name: Download Go Modules
+        run: go mod download
+      - name: Smoke Test Upgrade Path
```

Comment:
Isn't the repo still going to be on LATEST_RELEASE? Where's the upgrade occurring?

Reply:
This doesn't actually test the upgrade yet, it is just testing that the old version of the exemplar still works with the current FTL.
While this is a start I think we should enhance it as per my comment above. I don't think we need an external cluster to test this, we should be able to adapt the existing kube tests to perform an upgrade.

Reply:
I hit some kube snags trying to do this, but I'll dig in some more. Also, it seems as though the kube integration test support is for Go only atm